pkg/gatewayserver/gatewayserver.go

```diff
@@ -438,6 +440,10 @@ func (gs *GatewayServer) Connect(ctx context.Context, frontend io.Frontend, ids
 	logger.Info("Connected")
 	go gs.handleUpstream(connEntry)
 
+	if gtw.UpdateLocationFromStatus {
+		go gs.handleLocationUpdates(connEntry)
```

Comment:
Did you consider combining this goroutine with that of updating the connection stats?

Reply:
These two goroutines may be totally unrelated (one updates Redis, other updates GS + IS). Location updates may also be disabled per-gateway, whereas stats updates are always going to happen. Finally, it would be really hard to maintain two separate versions, merging every time one changes could be a serious cause of troubles in the long run.

Reply:
Still, eager to test out how that would work. Should we add an issue to the backlog for that?

Reply:
>These two goroutines may be totally unrelated (one updates Redis, other updates GS + IS).

They are related in the sense that there's a non-hot-path goroutine per gateway connection, that updates persistent state about a gateway. Also, these tasks don't bite each other.

>Still, eager to test out how that would work. Should we add an issue to the backlog for that?

Let's focus on adding functionality. I'm just trying to avoid spinning up a goroutine per gateway connection for each new feature.

Reply:
Agreed, if we are in no hurry for this particular feature, let's do this. Waiting for #2035 to be merged, then I will rebase and merge the two goroutines.

A simple solution would be to have a single `gatewayStatusChanged` channel, which would replace the `locationChanged` and the `statsChanged` channels.

Reply:
I think having separate channels is fine, you can select on both. You just need to figure out something smart for the debounce. An `After` also spins up a goroutine so that ain't gonna help.

Don't spend too much time on this. Sleeping goroutines is not a big problem.